### DISCUSSION TITLE
apple authentication

### DISCUSSION THREAD

**Discussion by dongri727** [serverpod/serverpod#2197] [Q&A]
I am working on apple authentication.
In the documentation, it says that there is no specific configuration on the Server side, do I need to configure anything with the apple key or key ID?
Currently, the apple button appears, I can set it up, it asks me if I want to sign in, I choose to continue with password, and when I enter the password, it is loading all the time and does not go any further. It is registered in the database as unknown.

**Comment by bfemmer** [comment]
There are platform-specific setup steps you need to accomplish in XCode (you really didn't mention this, so I'll start here). Take a look at [this article](https://medium.com/serverpod/integrating-apple-sign-in-with-serverpod-authentication-part-3-f5a49d006800) (which is a helpful link referenced from the [Apple auth provider documentation page](https://docs.serverpod.dev/concepts/authentication/providers/apple)) and let us know if this gets you going. I'm assuming you're running 1.2.6, but it would be helpful in a subsequent reply to reference the version of Serverpod your running if this is not the case. Good luck!
